### PR TITLE
[v6r15] Moved base security modules to using DErrno

### PR DESCRIPTION
**Pull request by fstagni**
And various minor stuff

### CODE REVIEW

**Comment by fstagni** [comment]
@atsareg @chaen : I quickly realized that adding error codes to DErrno, like I've done here, is sensitive to possibly disruptive changes if this is done in parallel by several users. We should pay special attention when these are merged.

File: Core/Security/ProxyFile.py [hunk 1]
```diff
@@ -101,14 +105,14 @@ def multiProxyArgument( proxy = False ):
     if not proxy:
       proxyLoc = getProxyLocation()
       if not proxyLoc:
-        return S_ERROR( "Can't find proxy" )
+        return DError( DErrno.EPROXYFIND )
```
Comment (atsareg) [inline Core/Security/ProxyFile.py]: Do we really need a special error for all the proxy related cases? Just a general EPROXY with a particular error message can be enough. I am a bit scared by the rapid proliferation of the error codes

File: Core/Security/X509Certificate.py [hunk 2]
```diff
@@ -70,7 +64,7 @@ def hasExpired( self ):
     Return: S_OK( True/False )/S_ERROR
     """
     if not self.__valid:
-      return S_ERROR( "No certificate loaded" )
+      return DError( DErrno.ENOCERT )
```
Comment (atsareg) [inline Core/Security/X509Certificate.py]: ECERTLOAD ? Or just a general ECERT ?

File: Core/Utilities/DErrno.py [hunk 3]
```diff
@@ -56,14 +56,37 @@
...
-# DMS/StorageManagement
+ECTMPF = 1007
+EOF = 1008
+ERF = 1009
+EWF = 1010
+ESPF = 1011
+
+# ## Core (11XX)
+# Certificates and Proxy: 0X
+EX509 = 1100
```
Comment (atsareg) [inline Core/Utilities/DErrno.py]: I think al those 10 error codes can be replaced by just EX509

**Comment by atsareg** [comment]
I agree that we can quickly have a mess with these error codes trying to sort out another mess. I would suggest that in general we introduce new error codes with a more specific meaning only when we have a real case that we have to distinguish them. But to begin with, let's introduce just a bare minimum.

File: WorkloadManagementSystem/Executor/JobScheduling.py [hunk 4]
```diff
@@ -221,10 +217,13 @@ def optimizeJob( self, jid, jobState ):
...
-  def __applySiteFilter( self, sites, active = False, banned = False ):
+  def _applySiteFilter( self, sites, banned = False ):
+    """ Filters out banned sites
+    """
+    if not sites:
+      return sites
+
     filtered = set( sites )
-    if active and isinstance( active, ( list, set, dict ) ):
-      filtered &= set( active )
```
Comment (atsareg) [inline WorkloadManagementSystem/Executor/JobScheduling.py]: Why this check is dropped ? Do you want sites not in mask to pass through ?

**Comment by fstagni** [comment]
Yes, on one side I agree with you. On the other side, for debugging purpose, we should try and be precise, otherwise if we end up putting several error types under the same one, we lose the advantage of the error type propagation that DError introduces.

**Comment by atsareg** [comment]
IIUC you can always add a more precise message string to the error code, this is what you can use for debugging.

**Comment by fstagni** [comment]
Sure sure. Just, I believe that if there's a place where we need to distinguish between errors with different error codes, than this has to be the Core, I agree that at the application level, e.g. in the TransformationSystem, we should restrict the number of error codes. I don't think that for this specific case, putting all the errors under the general umbrella of "EX509" would be beneficial. Anyway I am open for some changes, we are learning here.

**Comment by atsareg** [comment]
This branch has conflicts that must be resolved. Please, do

**Comment by fstagni** [comment]
PR is merg-able. I've also committed a few more.

**Pull request merged**